functions.sh

```diff
@@ -3,7 +3,7 @@
 # NOTE: These functions are used in the initramfs, so they must be ash/busybox compatible!
 
 info() {
-    echo "$@" 1>&2
+    grep debug /proc/cmdline && echo "$@" 1>&2
```

Comment:
Open to other options here, I just wanted something that wouldn't print by default

Reply:
This should still print if not running in the initramfs, so you need some other check for that. Maybe just have a QUIET variable and set it in the initramfs script if !debug?

Reply:
@Conan-Kudo do we have something we can use to check if we're running under dracut?

Reply:
You should check for `rd.debug` to invoke debug output. That's the standardized option among initramfs tools for this. Alternatively, you can check for `quiet` to suppress them instead.

If you wanted to check for Dracut specifically, the presence of `/lib/dracut-lib.sh` will tell you that. I don't recommend this though, since you want this to work with all initramfs tools (dracut, mkinitcpio, etc.)

Reply:
Of course, another option is that Asahi Linux distribution also uses dracut and then other initramfs tools don't matter. :)

Reply:
Updated to implement the logic suggested by @Conan-Kudo

Reply:
This still has the problem that it suppresses output with quiet mode enabled when you are using functions.sh from standalone scripts, which it shouldn't.

Reply:
The new implementation shouldn't, unless I missed something (GH is weird and this whole thread is under an "outdated" version, I'll mark it as resolved to avoid confusion).